### DISCUSSION TITLE
Optimization loop failed

### DISCUSSION THREAD

**Discussion by janiceylau** [talmolab/sleap#1220] [Help!]
Hello! After I installed SLEAP from source, I got the following error message: 

`W tensorflow/core/common_runtime/gpu/gpu_device.cc:1850] Cannot dlopen some GPU libraries. Please make sure the missing libraries mentioned above are installed properly if you would like to use GPU. Follow the guide at https://www.tensorflow.org/install/gpu for how to download and setup the required libraries for your platform.
Skipping registering GPU devices...
GPUs: None detected.`

So I followed the suggestion at the installation page and downgraded the tensorflow version to 2.6.3. But now there is another problem when I trained the model: 
![sleap_error](https://user-images.githubusercontent.com/102581787/225089849-b15411da-aa38-4cef-9583-486c0f9ee02b.png)

What should I do to solve this problem? Thanks very much for your help!

**Comment by roomrys** [comment]
Hi @janiceylau,

This could be a problem with allowing too much memory allocation for the GPU as seen [here](https://discuss.tensorflow.org/t/optimization-loop-failed-cancelled-operation-was-cancelled/1524/29). We should automatically be disabling pre-allocation (to avoid this error) IF a GPU is detected, see line 1992:
https://github.com/talmolab/sleap/blob/465ac074706b66ab0a74a59d7c75ac0a544c5df7/sleap/nn/training.py#L1960-L1996
Do you mind pasting you logs from the start of training (or even starting at Happy SLEAPing is good). I think your GPU might not be detected? even with downgrading tensorflow, but the logs will verify this.

Thanks,
Liezl
